tests/e2e/operator/multiRegion/cockroachdb_multi_region_e2e_test.go

```diff
@@ -0,0 +1,407 @@
+
+	// Validate CockroachDB functionality in each cluster.
+	for _, cluster := range clusters {
+		if _, ok := rawConfig.Contexts[cluster]; !ok {
+			t.Fatal()
+		}
+		rawConfig.CurrentContext = cluster
+		// Validate CockroachDB cluster.
+		r.ValidateCRDB(t, cluster)
+	}
+	// Validate Multi-region setup.
+	r.ValidateMultiRegionSetup(t)
+}
+
+// TestHelmUpgrade will upgrade the existing charts in multiple regions,
+// and verifies the CockroachDB health in multi-region.
+func (r *multiRegion) TestHelmUpgrade(t *testing.T) {
+	var corednsClusterOptions = make(map[string]coredns.CoreDNSClusterOption)
+
+	// Set up multi-region k3d infra.
+	r.SetUpInfra(t, corednsClusterOptions)
+
+	var clusters = operator.Clusters[:len(r.Clients)]
+
+	// Creating random namespace for each region.
+	for _, cluster := range clusters {
+		r.Namespace[cluster] = fmt.Sprintf("%s-%s", operator.Namespace, strings.ToLower(random.UniqueId()))
+	}
+
+	// Cleanup resources.
+	defer r.CleanupResources(t)
+
+	// Create CA certificate.
+	err := r.CreateCACertificate(t)
+	require.NoError(t, err)
+
+	defer r.CleanUpCACertificate(t)
+
+	// Apply operator, CockroachDB charts on each cluster.
+	for i, cluster := range clusters {
+		r.InstallCharts(t, cluster, i)
+	}
+
+	// Get current context name.
+	kubeConfig, rawConfig := r.GetCurrentContext(t)
+
+	// Validate CockroachDB functionality in each cluster.
+	for _, cluster := range clusters {
+		if _, ok := rawConfig.Contexts[cluster]; !ok {
+			t.Fatal()
+		}
+		rawConfig.CurrentContext = cluster
+		// Validate CockroachDB cluster.
+		r.ValidateCRDB(t, cluster)
+	}
+
+	// Get helm chart paths.
+	helmChartPath, _, err := operator.HelmChartPaths()
+	require.NoError(t, err)
+	for _, cluster := range clusters {
+		kubectlOptions := k8s.NewKubectlOptions(cluster, kubeConfig, r.Namespace[cluster])
+		options := &helm.Options{
+			KubectlOptions: kubectlOptions,
+			ExtraArgs: map[string][]string{
+				"upgrade": {"--reuse-values", "--set", fmt.Sprintf("operator.resources.requests.cpu=%s", "100m")},
+			},
+		}
+		// Apply Helm upgrade with updated values.
+		helm.Upgrade(t, options, helmChartPath, operator.ReleaseName)
+
+		// Get the initial timestamp of the pods before the upgrade.
+		pods := k8s.ListPods(t, kubectlOptions, metav1.ListOptions{
+			LabelSelector: operator.LabelSelector,
+		})
+		require.True(t, len(pods) > 0)
+
+		// Capture the creation timestamp of the last pod.
+		initialTimestamp := pods[0].CreationTimestamp.Time
+
+		// Verify if the pods are restarted after helm upgrade.
+		err = r.VerifyHelmUpgrade(t, initialTimestamp, kubectlOptions)
+		require.NoError(t, err)
+
+		crdbCluster := testutil.CockroachCluster{
+			DesiredNodes: r.NodeCount,
+		}
+		testutil.RequireCRDBClusterToBeReadyTimeout(t, kubectlOptions, crdbCluster, 600*time.Second)
+		pods = k8s.ListPods(t, kubectlOptions, metav1.ListOptions{
+			LabelSelector: operator.LabelSelector,
+		})
+		require.True(t, len(pods) > 0)
```

Comment:
Can we check that the number of ready pods is equal to the number of expected crdb nodes for the region? Since we're already listing pods and checking readiness in `RequireCRDBClusterToBeReadyTimeout`, maybe we can update that function to assert that we have the expected number of pods.